services/rpcfilters/hash_filter.go

```diff
@@ -0,0 +1,46 @@
+package rpcfilters
+
+import (
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+)
+
+type hashFilter struct {
+	hashes []common.Hash
+	mu     sync.Mutex
+	done   chan struct{}
+	timer  *time.Timer
+}
+
+// add adds a hash to the hashFilter
+func (f *hashFilter) add(data interface{}) {
+	hash := data.(common.Hash)
```

Comment:
shouldn't we have `if hash, ok := data.(common.Hash); ok` here so we never crash on type conversions?